TestPersistentVolumeClaimBinder flake

Seen in https://app.shippable.com/builds/56001b051a22100c00d15f6e running against https://github.com/kubernetes/kubernetes/pull/13835 (which had no API server changes)
```
W0921 15:45:04.798511   20405 controller.go:212] Resetting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0 0001-01-01 00:00:00 +0000 UTC   map[] map[]} [{[{172.17.5.229 }] [] [{ 6443 TCP}]}]}
E0921 15:45:04.858525   20405 repair.go:54] unable to persist the updated port allocations: 100: Key not found (/kubernetes.io) [159]
E0921 15:45:04.862195   20405 repair.go:69] unable to persist the updated service IP allocations: 100: Key not found (/kubernetes.io) [159]
E0921 15:45:05.041586   20405 persistentvolume_claim_binder_controller.go:132] PVClaimBinder could not add claim claim03: A volume match does not exist for persistent claim: claim03
--- FAIL: TestPersistentVolumeClaimBinder (0.55 seconds)
	persistent_volumes_test.go:109: Expected claim to be bound: &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:claim02 GenerateName: Namespace:default SelfLink:/api/v1/namespaces/default/persistentvolumeclaims/claim02 UID:ba75eb13-6077-11e5-a1f1-0242ac1105e5 ResourceVersion:166 Generation:0 CreationTimestamp:2015-09-21 15:45:05 +0000 UTC DeletionTimestamp: DeletionGracePeriodSeconds: Labels:map[] Annotations:map[]} Spec:{AccessModes:[ReadOnlyMany ReadWriteOnce ReadWriteMany] Resources:{Limits:map[] Requests:map[storage:{Amount:5000000000.000 Format:DecimalSI}]} VolumeName:} Status:{Phase:Pending AccessModes:[] Capacity:map[]}}
W0921 15:45:05.195558   20405 master.go:261] Network range for service cluster IPs is unspecified. Defaulting to 10.0.0.0/24.
I0921 15:45:05.195715   20405 master.go:287] Node port range unspecified. Defaulting to 30000-32767.
I0921 15:45:05.196348   20405 master.go:309] Will report 172.17.5.229 as public IP address.
```

cc @markturansky

--- a/test/integration/persistent_volumes_test.go
+++ b/test/integration/persistent_volumes_test.go
@@ -86,16 +86,18 @@ func TestPersistentVolumeClaimBinder(t *testing.T) {
 	}
 	defer watch.Stop()
 
-	boundCount := 0
-	expectedBoundCount := 2
-	for {
+	// Wait for claim01 and claim02 to become bound
+	claim01Pending := true
+	claim02Pending := true
+	for claim01Pending || claim02Pending {
 		event := <-watch.ResultChan()
 		claim := event.Object.(*api.PersistentVolumeClaim)
-		if claim.Spec.VolumeName != "" {
-			boundCount++
-		}
-		if boundCount == expectedBoundCount {
-			break
+		if claim.Spec.VolumeName != "" && claim.Status.Phase != "Bound" {
+			if claim.Name == "claim01" {
+				claim01Pending = false
+			} else if claim.Name == "claim02" {
+				claim02Pending = false
+			}
 		}
 	}
 

